File: solutions/java/high-scores/1/src/main/java/HighScores.java
import java.util.ArrayList;
import java.util.List;

class HighScores {

    private final List<Integer> highScores;

    public HighScores(List<Integer> highScores) {
        this.highScores = highScores;
    }

    List<Integer> scores() {
        return highScores;
    }

    Integer latest() {
        return highScores.get(highScores.size() - 1);
    }

    Integer personalBest() {
        Integer max = 0;
        for (Integer curr : highScores) {
            if (curr > max) {
                max = curr;
            }
        }
        return max;
    }

    List<Integer> personalTopThree() {
        List<Integer> copy = new ArrayList<>(highScores);

        copy.sort((a, b) -> b.compareTo(a));

        return copy.subList(0, Math.min(copy.size(), 3));
    }

}

File: solutions/csharp/high-scores/1/HighScores.cs
public class HighScores(List<int> list)
{
    private List<int> _highScores = list;

    public List<int> Scores()
    {
        return _highScores;
    }

    public int Latest()
    {
        return _highScores.Last();
    }

    public int PersonalBest()
    {
        return _highScores.Max();
    }

    public List<int> PersonalTopThree()
    {
        var copy = new List<int>(_highScores);
        copy.Sort((a, b) => b.CompareTo(a));
        
        return copy.Take(3).ToList();
    }
}